src/syscall.rs

```diff
@@ -451,6 +451,17 @@ impl<'c> SyscallContext<'c> {
         }
     }
 
+    pub extern "C" fn get_blob_hash_at_index(&mut self, index: &U256, blobhash: &mut U256) {
+        *blobhash = usize::try_from(index.lo).map_or(U256::default(), |idx| {
+            self.env
+                .tx
+                .blob_hashes
+                .get(idx)
+                .cloned()
+                .map_or(U256::default(), |x| U256::from_be_bytes(x.into()))
+        });
```

Comment:
This is a bit confusing. Generally, we have a chain of transformations (`map`, `and`, etc.), and at the end a single `unwrap`. In this case:

```suggestion
        *blobhash = usize::try_from(index.lo).and_then(|idx| {
            self.env
                .tx
                .blob_hashes
                .get(idx)
        })
        .map(U256::from_be_bytes)
        .unwrap_or_default();
```

Reply:
Yep, that's way better. Thanks for the idea.